src/modelinstance.hpp

```diff
@@ -33,11 +34,16 @@
 
 #include "customloaderconfig.hpp"
 #include "customloaderinterface.hpp"
+#include "deserialization.hpp"
+#include "executingstreamidguard.hpp"
 #include "modelchangesubscription.hpp"
 #include "modelconfig.hpp"
 #include "modelinstanceunloadguard.hpp"
 #include "modelversionstatus.hpp"
 #include "ovinferrequestsqueue.hpp"
+#include "prediction_service_utils.hpp"
+#include "processing_spec.hpp"
+#include "serialization.hpp"
```

Comment:
Do we need serialization & deserialization headers here?

Reply:
Yes. ModelInstance now contains infer(...) method that does **deserialization**, inference and **serialization**.